### DISCUSSION TITLE
지금 작성한 Next.js 코드가 App Routing인가요 Page Routing인가요??

### DISCUSSION THREAD

**Discussion by falconlee236** [manito42/FrontEnd#30] [Q&A] [answered]
[https://nextjs.org/docs/getting-started/project-structure](url)
여기에 있는 공식 문서를 읽고 있는데, 우리 프로젝트가 App routing으로 짜여있는지, Page routing으로 짜여있는지 잘 모르겠네요
아직 Next.js초보라서 궁금한게 참 많은것 같습니다.

아 그리고 React에서는 index.js에서 부터 프로그램을 시작하는 것 같은데, 이 코드에서도 index.tsx에서부터 시작하는 것 같은데 그점이 맞나요??
지금 코드 구조 파악하면서 + Next.js 공식문서 읽기 + Redux 공식문서 읽기 해보려고 합니다.

구조는 여기까지 이해한 상태입니다!! 틀린점 있으면 지적해주세요
![image](https://github.com/manito42/FrontEnd/assets/52999093/e613a7c3-f309-402d-bea0-acfdcafe948d)

**Comment by JuneParkCode** [comment]
@9utty @nyeoni

**Comment by 9utty** [comment]
일단 폴더구조를 보시면 pages/ 로 시작하는 NextJS구조는 page route입니다. app/으로 시작하는 폴더구조가 app route라고 생각하시면 됩니다!

고로 우리가 사용하는 route방식은 page route입니다!

**Comment by 9utty** [comment]
react와 다른점은 사실 없습니다! 실제로 index.tsx로 개발자가 작성하는게 index.js로 컴파일이 됩니다! (트랜스파일러)개념입니다! react와 다른 점이라면 한가지 더 있는데 page route일땐 url/<filePath>로 라우팅이 강제로 잡힙니다! 그래서 지금 프론트 폴더구조를 보시면 pages/profile/index.tsx 와 비슷한 구조를 가집니다.

**Comment by 9utty** [comment, marked as answer]
pages route에서 처음 42manito.com 을 접속한다고 가정을 한다면 pages/index.tsx에서 가장 먼저 라우팅이 됩니다! home이라고 생각하시면 될 것 같습니다!

그래서 지금 플로우 작성 하신 것 확인 했는데 제대로 파악하신 것 같습니다! 너무 잘하셔서 놀랬습니다...!

다른 질문있으시면 디스커션에 또 적어주세요!
@falconlee236

**Reply by falconlee236** [reply]
이런 기능 처음 만져봐서 이것저것 눌러보다가 answer 하는 사람의 이름이 바뀌었네요
담부턴 신중히 누르겠습니다!!
답변 감사합니다 이제 page route에 대해서 계속 알아보러 갈게요 👍
